pkg/backup/backupdest/backup_index_test.go

```diff
@@ -481,6 +483,308 @@ func TestIndexExists(t *testing.T) {
+		require.NoError(t, err)
+		if aEnd.Before(bEnd) {
+			return -1
+		} else if aEnd.After(bEnd) {
+			return 1
+		}
+		// end times are equal, compare start times
+		if bStart.Before(aStart) {
+			return 1
+		} else {
+			return -1
+		}
+	}), "indexes are not sorted by end time and then start time")
+}
+
+func TestGetBackupTreeIndexMetadata(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
+
+	ctx := context.Background()
+	st := cluster.MakeTestingClusterSettingsWithVersions(
+		clusterversion.Latest.Version(),
+		clusterversion.Latest.Version(),
+		true,
+	)
+	execCfg := &sql.ExecutorConfig{Settings: st}
+
+	const collectionURI = "nodelocal://1/backup"
+	dir, dirCleanupFn := testutils.TempDir(t)
+	defer dirCleanupFn()
+	externalStorage, err := cloud.ExternalStorageFromURI(
+		ctx,
+		collectionURI,
+		base.ExternalIODirConfig{},
+		st,
+		blobs.TestBlobServiceClient(dir),
+		username.RootUserName(),
+		nil, /* db */
+		nil, /* limiters */
+		cloud.NilMetrics,
+	)
+	require.NoError(t, err)
+	storageFactory := func(
+		_ context.Context, _ string, _ username.SQLUsername, _ ...cloud.ExternalStorageOption,
+	) (cloud.ExternalStorage, error) {
+		return externalStorage, nil
+	}
+
+	writeIndex := func(
+		t *testing.T, subdirEnd, start, end int,
+	) {
+		t.Helper()
+		subdirTS := time.Date(2025, 8, 12, subdirEnd, 0, 0, 0, time.UTC)
+		startTS := hlc.Timestamp{WallTime: time.Date(2025, 8, 12, start, 0, 0, 0, time.UTC).UnixNano()}
+		if start == 0 {
+			startTS = hlc.Timestamp{}
+		}
+		endTS := hlc.Timestamp{WallTime: time.Date(2025, 8, 12, end, 0, 0, 0, time.UTC).UnixNano()}
+		subdir := subdirTS.Format(backupbase.DateBasedIntoFolderName)
+
+		details := jobspb.BackupDetails{
+			Destination: jobspb.BackupDetails_Destination{
+				To:     []string{collectionURI},
+				Subdir: subdir,
+			},
+			StartTime:     startTS,
+			EndTime:       endTS,
+			CollectionURI: collectionURI,
+			// This test doesn't look at the URI stored in the index metadata, so it
+			// doesn't need to be accurate to the exact path differences between full
+			// and incremental backups. We can just set URI to something that looks
+			// valid.
+			URI: collectionURI + subdir,
+		}
+		require.NoError(t, WriteBackupIndexMetadata(
+			ctx, execCfg, username.RootUserName(), storageFactory, details,
+		))
+	}
+
+	// Indexes represents the set of index files in the backup collection that the
+	// test cases will be running against. We use ints to represent hours within a
+	// single day to simplify test data.
+	indexes := []struct {
+		// endTime of the full backup.
+		subdirEnd int
+		// start and end times of all backups in the chain, including the full.
+		times [][2]int
+	}{
+		{
+			subdirEnd: 2,
+			times:     [][2]int{{0, 2}, {2, 4}, {4, 6}, {6, 8}},
+		},
+		// Contains compacted backup covering t=10-14
+		{
+			subdirEnd: 10,
+			times:     [][2]int{{0, 10}, {10, 11}, {11, 12}, {12, 14}, {10, 14}, {14, 16}, {16, 18}},
```

Comment:
nit: does this case need 8 backups to test a single compaction? always good to create the simplest possible test case for what you're testing.